### ISSUE TITLE
Simplify election process

### ISSUE THREAD

**Issue by tobie** [openjs-foundation/cross-project-council#961]
There's a lot of complexity and bureaucracy involved in electing voting members, and it seems that being a voting members is essentially useful for three things:

1. voting on who gets to be a voting member _(yes, that's kind of absurd)_,
2. voting on a decision when the CPC is unable to reach consensus _(I don't think that ever happened)_, and
3. looking good on a resume _(I don't want to dismiss that altogether, it actually allows people to carve out time on their day job to participate)_.

This seems suboptimal.

What if you got a nice title after joining the CPC for a little while ("Active Member" or something) and all active members would get to vote in case of a consensus issue (maybe with a per project or per company cap) and we can get rid of elections altogether?

**Comment by mcollina** [comment]
While I agree on the sentiment, I disagree on the solution.

The CPC is structured to
1. have no barrier to entry, enabling whoever wants participate to join
2. give stability guarantees to impact projects

The system was designed to avoid a takeover of the CPC. The voting members guarantee such stability, more or less giving impact projects a veto on policies they would go against their interests.

I'm happy to provide a bit more context in private.

---

The model that @tobie is describing looks a lot like the TSC model of Node.js, which we wanted to avoid when defining how the CPC would work.

**Comment by tobie** [comment]
I’m totally open to other solutions, btw. I literally made that solution up yesterday, as I was filing the issue.

That said, a hostile takeover is highly improbable _and_ the board could overrule it by changing the bylaws.

**Comment by ovflowd** [comment]
Agree with @mcollina I could see potential situations, where if non-voting members get to elect completely strangers to the CPC (Because the voting process for adding regular members is vulnerable to that), it could possibly create the risks for a takeover of the CPC.

This is something that _kinda_ "happened" on the GNOME Foundation Board of Directors when they voted that non-Foundation members can be part of the Board of Directors (a little bit different, yet similar).

It truly feels like the current mechanisms are in place to protect exactly this from happening.

Off-topic: Not nit-picking, but it kinda feels harsh to say that Voting members are voting members for the sake of their resumé... Not sure what was the intention with that comment, I possibly misunderstood it 😅

**Comment by tobie** [comment]
> This is something that _kinda_ "happened" on the GNOME Foundation Board of Directors when they voted that non-Foundation members can be elected to become part of the Board of Directors (a little bit different, yet similar).

Yeah, I do want to point out, that there's a board sitting on top of the CPC who owns the bylaws and could essentially kill the CPC or remove it altogether if necessary, so there's already a pretty solid (if cumbersome) backstop measure in case of a real crisis.

**Comment by ovflowd** [comment]
> Yeah, I do want to point out, that there's a board sitting on top of the CPC who owns the bylaws and could essentially kill the CPC or remove it altogether if necessary, so there's already a pretty solid (if cumbersome) backstop measure in case of a real crisis.

But that be an extreme action. I think if we can prevent it from reaching there, then we should keep mechanisms that allow us to self-moderate so that such circumstances don't happen... imho.

**Comment by tobie** [comment]
It seems there's agreement on what we're trying to achieve: remove red tape while keeping the CPC protected from hostile takeover.

Would y'all be willing to attempt to define the threat mechanisms and suggest solutions that would mitigate those while allowing us to reduce the bureaucracy?

**Comment by tobie** [comment]
> But that be an extreme action. I think if we can prevent it from reaching there, then we should keep mechanisms that allow us to self-moderate so that such circumstances don't happen... imho.

This is a classic impact/probability risk assessment problem. You want to reduce the impact of rare but critical events, but not to the point that doing so creates more important risk elsewhere (such as reducing operational agility by making everything too cumbersome for example).

Maybe this bureaucratic layer is the right tradeoff. Maybe its over-conservative and there's a more nimble tradeoff that reduces our overall risk, not just this one.

**Comment by mcollina** [comment]
I would consider getting the Board involved in projects matter would be a _massive_ failure for the CPC.

Here is an alternative setup: keep the voting members from Impact projects but drop the at-large project representative and the one elected by regular members, essentially removing all the elections from within the CPC.

**Comment by eemeli** [comment]
> Here is an alternative setup: keep the voting members from Impact projects but drop the at-large project representative and the one elected by regular members, essentially removing all the elections from within the CPC.

This would disenfranchise at-large projects completely, which I do not support.

My alternative proposal here would be to drop the elected at-large project and regular-member voting representatives, and instead allow all at-large projects to nominate one voting representative each. This would similarly remove all voting-member elections from the CPC.

**Comment by tobie** [comment]
> I would consider getting the Board involved in projects matter would be a massive failure for the CPC.

That's a great risk upper bound to work with.

> My alternative proposal here would be to drop the elected at-large project and regular-member voting representatives, and instead allow all at-large projects to nominate one voting representative each. This would similarly remove all voting-member elections from the CPC.

I really like @eemeli's proposal for its potential of increasing involvement of project in the foundation's activities.

However, it does dramatically change the balance between _Impact_ and _At Large_ projects compared to what it is now.

Currently the ratio is of 11 voting members for _Impact_ projects for 2 _At Large_ reps. @eemeli's proposal would change that _Impact_:_At Large_ ratio from 11:2 to 11:29 (if we count incubating projects), or more roughly from a 5:1 ratio to a 1:3.

That's quite a dramatic change and feels undesirable. What if we tied voting to projects rather than representatives and gave _Impact_ projects more weight. Giving each _Impact_ project 5 votes, for example would gives us the following _Impact_:_At Large_ ratio currently: 30:29, which I find quite appealing, actually.

We could even imagine that the number of voices of _Impact_ projects could be tuned on a regular basis to obtain an agreed-upon equilibrium.

**Comment by ovflowd** [comment]
@tobie would it make sense to increase the amount of voters per impact project if we increase the participation of at large projects?

**Comment by mcollina** [comment]
I like the direction we are going but I'm afraid we would fail to have a quorum most of the time, and we should adjust our rules somehow to lower the quorum requirements. 

@eemeli you have a bit more pulse on the at-large projects than me: what's the appetite for more involvement?

As a side note, collecting _60_ votes is hard work. Maybe we can have 1 representative per project, but the impact representatives have a 5x weight?

**Comment by ovflowd** [comment]
Yeah in that sense consensus can definitely be hard. +1 then for impact projects having 5x weight

**Comment by joesepi** [comment]
Honestly, I think this is a non-issue. I think it maybe it felt like an issue because this year was very bumpy and got confusing at times because of a few somewhat large errors in the process.

We essentially have three voting occasions. (see below) Those occasions are separated into three so that people could focus on one of the three higher level positions individually (CPC chair, CPC Board Seat and Community Board Seat) and we grouped the At large and  Regular member vote with Secondary Board seat at one time to maximize awareness around OpenJS world in 2021.

Personally, I dont think this is too much voting. And I dont think this is too many or two little voting members. I think the representation is fair and makes sense to me.

What I do think is that folks who run for At large voting member positions (up to 2) and regular member voting positions (up to 2) should be expected to do more to represent their constituencies and thus validating the positions. This is not a knock on current or previous members. I just think these roles could be defined a bit better and expectations could be set a bit better.

And I agree with others in saying that we set the structure up for good reasons. There have been issues in the community in the past that we are guarding against but again, I dont really think there is _too much voting_ :)

Voting schedule
- APR 1 - CPC Chair election - term ends April 30
- FEB 1 - Director 1 election - term ends February 28
- JUL 1 - Director 2 election - term ends July 31
- JUL 1 - CPC At Large voting member election - term ends July 31
- JUL 1 - CPC Regular member voting member election - term ends July 31

**Comment by tobie** [comment]
> As a side note, collecting 60 votes is hard work. Maybe we can have 1 representative per project, but the impact representatives have a 5x weight?

Yes. That was what I had in mind.

**Comment by tobie** [comment]
@joesepi we meet roughly 25 times a year. Each election requires at the very least the following:

- announcing that the nominations are opened during a call,
- announcing the nominees during a second call and kickstarting voting, and
- announcing the result of the elections.

So at the very minimum we have 9 calls out 25 were we are actively dealing with voting. That's more than one out of three calls. That's a lot. And frankly feels mostly unnecessary; it's pretty much always the same people in the same roles anyway.

Your point about At Large reps interacting more with the community is well taken, and as you know has been a real concern of mine (see: #761). It is telling, imho, that both @eemli and myself (the current At Large reps) are trying to find a solution that involves At Large maintainers being involved directly. To make this work would require At Large reps become community managers, and that's too big an ask from volunteers.

**Comment by eemeli** [comment]
> @mcollina:
> I like the direction we are going but I'm afraid we would fail to have a quorum most of the time, and we should adjust our rules somehow to lower the quorum requirements.

Umm, I don't think we have an explicit definition of quorum for voting members? The only textual reference to a quorum is here:
https://github.com/openjs-foundation/cross-project-council/blob/88d5e39c48c2aebd635dca3ddc35bba3e87178f0/governance/GOVERNANCE.md?plain=1#L200-L201

The context for how we ended up with that is https://github.com/openjs-foundation/cross-project-council/pull/205#discussion_r295374271, which doesn't really make it clear what the quorum definition actually is. That was in June 2019, and going back a step further we weren't sure at least in October 2018 what that definition was: https://github.com/openjs-foundation/cross-project-council/blob/88d5e39c48c2aebd635dca3ddc35bba3e87178f0/meetings/2018/2018-10-22.md?plain=1#L89-L92

It would seem that in #204 there's an implicit expectation that quorum means "half the voting members", but do we have that codified anywhere? If so, I can't find it.

> @mcollina:
> @eemeli you have a bit more pulse on the at-large projects than me: what's the appetite for more involvement?

Very little active interest. The CPC doesn't really do much that affects our at-large projects; the main benefit it brings is to have some of the processes and preparations ready in case Bad Things happen. I rather hope that with e.g. #953 we'll be able to do more for our projects, which might of course change this state of affairs.

> @tobie:
> So at the very minimum we have 9 calls out 25 were we are actively dealing with voting. That's more than one out of three calls. That's a lot. And frankly feels mostly unnecessary; it's pretty much always the same people in the same roles anyway.

I agree. We spend so much time on meta-level things in the CPC that we're limiting the scope of actual work that gets done. Each of us only has a limited bandwidth to give to OpenJS work, so reducing the baseline management tasks could have an outsized impact on our ability to deliver actual benefits to people and projects.

**Comment by tobie** [comment]
@eemeli wrote:
>  We spend so much time on meta-level things in the CPC that we're limiting the scope of actual work that gets done. Each of us only has a limited bandwidth to give to OpenJS work, so reducing the baseline management tasks could have an outsized impact on our ability to deliver actual benefits to people and projects.

👆 THIS 👆

**Comment by joesepi** [comment]
Hey @dylans - I just wanted to reply to your 
> * I cannot remember when we meet. I have a stale calendar entry that I can't seem to remove. And the meeting time is often in the prime time slot for meetings between teams in the US and Europe, which often causes conflicts.

Let me know if I can get you an updated calendar invite or you can do what I do and subscribe to the public calendar: https://calendar.openjsf.org. To your other point, perhaps we should look at other time slots to be more accommodating.

> * I get so many github notification emails that I rarely see the ones related to the CPC. For the calls I do make I rarely see the agenda prior to the call.

My routine is to jump into the github issues to quickly find the meeting issue with the agenda on Tuesday mornings.
https://github.com/openjs-foundation/cross-project-council/issues

> * We use a Google Doc for each call to track notes, which links to various GitHub issues, but I rarely remember where that Google Doc lives to see what I missed in a meeting.

The goal is always to get these meeting notes (google doc) committed to the repo. I'm trying to get better at doing that in a more timely manner. They live here: https://github.com/openjs-foundation/cross-project-council/tree/main/meetings

I hope this is helpful for you and anyone else who finds this comment. :)

**Comment by tobie** [comment]
This has been resolved in #1698
